sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentUtils.scala

```diff
@@ -210,13 +241,165 @@ object AssignmentUtils extends SQLConfHelper with CastSupport {
     }
   }
 
+  private def applyNestedFieldAssignments(
```

Comment:
Note: this is like applyFieldAssignments above, but recurses into nested structs